src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1441,6 +1441,74 @@ def accept_market_ordering_terms(cmd, urn=None, publisher=None, offer=None, plan
 # endregion
 
 
+def _terms_prepare(cmd, urn, publisher, offer, plan):
+    if urn:
+        if any([publisher, offer, plan]):
+            raise CLIError('usage error: If using --urn, do not use any of --plan, --offer, --publisher.')
+        terms = urn.split(':')
+        if len(terms) != 4:
+            raise CLIError('usage error: urn format is wrong. It should be in format of publisher:offer:sku:version.')
+        publisher, offer = terms[0], terms[1]
+        image = show_vm_image(cmd, urn)
+        if not image.plan:
+            raise CLIError("Image '%s' has no terms to accept." % urn)
+        plan = image.plan.name
+    else:
+        if not all([publisher, offer, plan]):
+            raise CLIError('usage error: Please provide --plan, --offer and --publisher, or --urn instead.')
+    return publisher, offer, plan
+
+
+def _accept_cancel_terms(cmd, urn, publisher, offer, plan, accept):
+    publisher, offer, plan = _terms_prepare(cmd, urn, publisher, offer, plan)
+    op = cf_vm_image_term(cmd.cli_ctx, '')
+    terms = op.get(publisher, offer, plan)
+    terms.accepted = accept
+    return op.create(publisher, offer, plan, terms)
+
+
+def accept_terms(cmd, urn=None, publisher=None, offer=None, plan=None):
+    """
+    Accept terms.
+    :param cmd:cmd
+    :param urn:URN, in format of 'publisher:offer:sku:version'. If specified, other argument values can be omitted
+    :param publisher:Image publisher
+    :param offer:Image offer
+    :param plan:Image billing plan
+    :return:
+    """
+    return _accept_cancel_terms(cmd, urn, publisher, offer, plan, True)
+
+
+def cancel_terms(cmd, urn=None, publisher=None, offer=None, plan=None):
+    """
+    Cancel terms.
+    :param cmd:cmd
+    :param urn:URN, in format of 'publisher:offer:sku:version'. If specified, other argument values can be omitted
```

Comment:
well. re-think about this. Same help information appear in three places.

Reply:
Yes. But it's not code. I can't use variables to concatenate doc.

Reply:
Updated